internal/graphql/model/operator_helper.go

```diff
@@ -206,20 +211,36 @@ func SubscribeToBroker(provider models.Provider, mesheryKubeClient *mesherykube.
 			}
 		}
 	}
-
+	endpoints = append(endpoints, endpoint)
 	// subscribing to nats
 	conn, err := nats.New(nats.Options{
-		URLS:           []string{endpoint},
+		URLS:           endpoints,
 		ConnectionName: "meshery",
 		Username:       "",
 		Password:       "",
 		ReconnectWait:  2 * time.Second,
 		MaxReconnect:   5,
 	})
 	// Hack for minikube based clusters
-	if err != nil {
+	if err != nil && conn == nil {
 		return endpoint, err
 	}
+	defer func() {
+		if conn == nil {
+			return
```

Comment:
Is connection state being tracked anywhere? If so, should this update that state?

Reply:
This connection is local to the function call SubscribeToBroker so it doesn't need to be tracked anywhere else. 
Just few lines below, you will find a DeepCopy method being invoked which actually copies this local instance of connection into globally maintained broker connection if no errors are encountered.